features/FEATURE_PSA/TARGET_TFM/TARGET_TFM_DUALCPU/src/tfm_mbed_boot.c

```diff
@@ -0,0 +1,58 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2020 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mbed_error.h"
+#include "tfm_multi_core_api.h"
+#include "tfm_ns_mailbox.h"
+#include "platform_multicore.h"
+#include "tfm_ns_interface.h"
+
+static struct ns_mailbox_queue_t ns_mailbox_queue;
+
+void mbed_main(void)
+{
+    /*
+     * In case the of Dual Core, we need to initialize the IPC, Mailbox
+     * NS interface after the RTOS has started to enable communication
+     * from Secure and Non-Secure cores.
+     */
+    int32_t ret;
+
+    ret = tfm_ns_wait_for_s_cpu_ready();
+    if (ret != PLATFORM_MAILBOX_SUCCESS) {
```

Comment:
This would require changes to TF-M as these files are imported as part of the integration. I could remove `PLATFORM_MAILBOX_SUCCESS` and check as `if(!ret)` instead.

Reply:
Yes, this requires changes to TF-M.

Reply:
On closer investigation - https://git.trustedfirmware.org/trusted-firmware-m.git/tree/app/main_ns.c#n90, seems like they never check the return type.

Have created - https://developer.trustedfirmware.org/T660

Reply:
It'd be quicker to fix TF-M directly.

Reply:
Could you please link to the TF-M issue, and explain what's wrong with the code as currently written, in a code comment here? We should ensure we don't keep this bad code here forever. Linking to the issue and noting that we can use TFM_SUCCESS after it is fixed will help us know when we can fix the code and why the code is currently not quite right.